### DISCUSSION TITLE
Run GitHub Action with Conditions

### DISCUSSION THREAD

**Discussion by vybhav72954** [community/community#25421] [Actions]
<p>I have a Repo, where people can upload their scripts, in which they solve a particular problem using more than one language.</p>
<p>The Scripts are segregated according to Languages. i.e. there are folders <code>Python</code>, <code>JavaScript</code>, <code>Go</code> etc and there are multiple subdirectories in them and they contain the script and a <code>README.md</code>.</p>
<p>I want to make a PyLint Action, so ta when someone starts a PR, the action tells me their PyLint score and I can decide whether to merge or not.</p>
<p>I want this action to be triggered only when someone Starts a PR and the commits have <code>.py</code> file (i.e. Language used Is python).</p>
<p>Can someone guide m please.</p>
<p>in <code>.yml</code> file I have</p>
<pre><code class="lang-auto">on:
  pull_request:
    branches:
      - master
    path:
      - '**.py'
</code></pre>
<p>This is not solving the issue.</p>
<p>Complete Workflow</p>
<pre><code class="lang-auto">name: PyLint Runner

on:
  pull_request:
    branches:
      - master
    path:
      - '**.py'

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v1
    - name: Set up Python 3.8
      uses: actions/setup-python@v1
      with:
        python-version: 3.8
    - name: Install Requirements
      run: |
        python -m pip install --upgrade pip
        pip install pylint
    - name: Run Pylint
      run: |
        python pyLint.py --path ./src --threshold 1.5
</code></pre>

**Comment by kingthorin** [comment, marked as answer]
<p>You could use an action like <code>futuratrepadeira/changed-files</code> to get the changed files.</p>
<p>You can then gather added or updated files for other steps, such as:</p>
<pre><code class="lang-auto">env:
        FILES: '${{ steps.files.outputs.files_updated }} ${{ steps.files.outputs.files_created }}'
</code></pre>
<p>Here’s a workflow that does link checks in added/changed markdown files that you could use as a model for your ointing requirements: <a href="https://github.com/OWASP/wstg/blob/559a2794ae26adaf9dc7187b3ae60ab161e1ca17/.github/workflows/md-link-check.yml" rel="noopener nofollow ugc">https://github.com/OWASP/wstg/blob/559a2794ae26adaf9dc7187b3ae60ab161e1ca17/.github/workflows/md-link-check.yml</a></p>
